lib/transaction/transaction.js

```diff
@@ -713,6 +713,21 @@ Transaction.prototype.addOutput = function(output) {
   return this;
 };
 
+
+/**
+ * Remove all outputs from the transaction.
+ *
+ * @return {Transaction} this, for chaining
+ */
+Transaction.prototype.clearOutputs = function() {
+  this.outputs = [];
+  this._clearSignatures();
+  this._outputAmount = undefined;
+  this._changeIndex = undefined;
+  return this;
```

Comment:
Shouldn't this also call `this._updateChangeOutput()`? Otherwise all the inputs will be miner's fee.